test/distributed/test_autograd.py

```diff
@@ -30,6 +30,7 @@
 
 def run_test_scatter_v(rank, world_size):
     os.environ["RANK"] = f"{rank}"
+    os.environ["LOCAL_RANK"] = f"{rank}"
```

Comment:
not sure if necessary, but tried a few things after observing several different hangs for local tests

Reply:
That's weird. The default is to set the local rank to `rank % gpus_per_node` if this env var is not defined. If that's failing, maybe there is some other issue in the setup process that's worth looking into?

Reply:
Personally, not really sure about that. I guess this is just left over from debugging something else. I'll remove these.

Reply:
checked these again, the issue was with LOCAL_RANK being set but being empty
in this case, `int(os.environ.get(...)` throws a type error and `initialize_env` is aborted, the manager tries unsuccessfully to `initialize_slurm` or `initialize_open_mpi`. As these are not even tried (because the "guarding" environment variables don't exist), `initialize` just returns without setting up anything. The first NCCL primitive then errors out with the default process group not having been initialized.

```        
if initialization_method is None:
            try:
                DistributedManager.initialize_env()
            except TypeError:
                if "SLURM_PROCID" in os.environ:
                    DistributedManager.initialize_slurm(port)
                elif "OMPI_COMM_WORLD_RANK" in os.environ:
                    DistributedManager.initialize_open_mpi(addr, port)
```

I have removed all LOCAL_RANK occurrences in all tests but also updated the DistributedManager to cover this case in a more robust fashion. An empty LOCAL_RANK now doesn't lead to things erroring out but to assuming the default of `rank % device_count`.

Reply:
Ah, it's about `LOCAL_RANK` being defined but empty. That makes sense.

> DistributedManager to cover this case in a more robust fashion

Yeah, I agree this is needed. Can you ensure that the changes you are introducing are not already covered in #475 or causes any conflicts with that PR?

Reply:
oh wow, this PR is still open? 
indeed, it introduces a "warn" in the "else"-branch which later leads to an exception afterwards
let's merge this PR first and i'll resolve the conflicts